source/material_model/rheology/elasticity.cc

```diff
@@ -262,12 +261,10 @@ namespace aspect
               for (unsigned int j=0; j < SymmetricTensor<2,dim>::n_independent_components; ++j)
                 stress_old[SymmetricTensor<2,dim>::unrolled_to_component_indices(j)] = in.composition[i][j];
 
-              // Average viscoelastic viscosity
-              const double average_viscoelastic_viscosity = out.viscosities[i];
-
-              // Fill elastic force outputs (See equation 30 in Moresi et al., 2003, J. Comp. Phys.)
-              force_out->elastic_force[i] = -1. * ( average_viscoelastic_viscosity / calculate_elastic_viscosity(average_elastic_shear_moduli[i]) * stress_old );
-
+              // The viscoelastic strain rate is needed only when the Newton method or the DCP method is used for solving.
+              elastic_out->elastic_force[i] = -out.viscosities[i] / calculate_elastic_viscosity(average_elastic_shear_moduli[i]) * stress_old;
+              if (Parameters<dim>::is_defect_correction(this->get_parameters().nonlinear_solver))
```

Comment:
Will this condition also capture the case for the full Newton solver?

Reply:
Yes. In fact, after restoring the RHS only the full Newton solver needs the viscoelastic strain rate now.